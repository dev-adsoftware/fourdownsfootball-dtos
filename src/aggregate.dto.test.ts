/* eslint-disable max-classes-per-file */

import { Aggregate, Event } from '.';

describe('given: aggregate data', () => {
  describe('when: I serialize the aggregate', () => {
    it('then: result was serialized aggregate data', async () => {
      const aggregate = new Aggregate().init({
        id: 'jest.id',
        sequence: '0',
        date: '2000-01-01T00:00:00.000Z',
        event: new Event().init({
          source: 'jest.source',
          type: 'jest.type',
          version: 'jest.version',
        }),
      });
      expect(aggregate.serialize()).toEqual(
        '{"id":"jest.id","sequence":"0","date":"2000-01-01T00:00:00.000Z","event":{"source":"jest.source","type":"jest.type","version":"jest.version"}}',
      );
    });
  });
});
