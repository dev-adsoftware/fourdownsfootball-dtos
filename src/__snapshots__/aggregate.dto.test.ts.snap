// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given: no aggregate data when: I create an aggregate dto then: dto threw expected error 1`] = `"Aggregate: id must be a string, sequence must be a number conforming to the specified constraints, date must be a Date instance"`;
